### PR TITLE
feat(example): Add a python example to codegen whole gemm.

### LINKED ISSUE #16: Add a Python example for whole GEMM.

### PR DESCRIPTION
**Pull request by KuangjuX**
resolved #16.

### CODE REVIEW

File: thriller-bindings/examples/gemm/gemm_g2r.py [hunk 1]
```diff
@@ -45,37 +47,35 @@
...
 
-    NodeA = pythriller.PyNode(rA)
-    NodeB = pythriller.PyNode(rB)
-    NodeAcc = pythriller.PyNode(acc)
+    NodeA = Node.tensor(rA)
+    NodeB = Node.tensor(rB)
+    NodeAcc = Node.tensor(acc)
 
-    GemmNode = pythriller.PyNode.gemm(NodeA, NodeB, NodeAcc)
+    GemmNode = Node.gemm(NodeA, NodeB, NodeAcc)
```
Comment (haruhi55) [inline thriller-bindings/examples/gemm/gemm_g2r.py:56]: I just have some questions due to my curiosity.

1. Does `Tensor(data)` also represent a node in the graph, functioning similarly to a "buffer"?

2. Intuitively, `gemm` is an operation with input and output. How does it differentiate between the input and output?

File: thriller-bindings/examples/gemm/gemm_g2r.py [hunk 2]
```diff
@@ -45,37 +47,35 @@
...
 
-    AccessMap = pythriller.AccessMap(
+    AccessMap = AccessMap(
         access_dims, [[[1]], [[0]]], [[0], [10]], [LoopIter])
 
-    EdgeA_Gemm = pythriller.PyEdge(NodeA, GemmNode)
-    EdgeB_GEMM = pythriller.PyEdge(NodeB, GemmNode)
-    EdgeGemm_Acc = pythriller.PyEdge(GemmNode, NodeAcc)
+    EdgeA_Gemm = Edge(NodeA, GemmNode)
+    EdgeB_GEMM = Edge(NodeB, GemmNode)
```
Comment (haruhi55) [inline thriller-bindings/examples/gemm/gemm_g2r.py:66]: `EdgeA_gemm` and `EdgeB_GEMM` have different naming style, is this intentional?

File: thriller-bindings/examples/gemm/whole_gemm.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
+'''
+Whole GEMM is an example of GEMM that utilized all mempry hierarchies
```
[the remark this answers is not in the document]
Comment (KuangjuX) [reply thriller-bindings/examples/gemm/whole_gemm.py]: fixed.

File: thriller-bindings/examples/gemm/gemm_g2r.py (hunk 2 above)
Comment (KuangjuX) [reply thriller-bindings/examples/gemm/gemm_g2r.py]: fixed.

File: thriller-bindings/examples/gemm/gemm_g2r.py (hunk 1 above)
Comment (KuangjuX) [reply thriller-bindings/examples/gemm/gemm_g2r.py:56]: 1. A `Tensor` is essentially a Buffer, which I have `renamed`:
```rust
#[pyclass(unsendable, module = "buffer", name = "Tensor")]
pub struct PyBuffer(pub Rc<Buffer>);
```
2. `gemm` takes three parameters: `a` and `b` as inputs, and `c` as the output:
```rust
    #[staticmethod]
    fn gemm(a: PyRef<PyNode>, b: PyRef<PyNode>, c: PyRef<PyNode>) -> Self {
        let access_map = AccessMap::new(0, vec![]);

        let node_a = Rc::clone(&a.0);
        let node_b = Rc::clone(&b.0);
        let node_c = Rc::clone(&c.0);

        let gemm = Gemm::new(vec![node_a, node_b], node_c, Rc::new(access_map));

        let node = ThrillerNode::new(ThrillerNodeInner::Op(Box::new(gemm)));

        PyNode(Rc::new(RefCell::new(node)))
    }
```

Comment (haruhi55) [reply thriller-bindings/examples/gemm/gemm_g2r.py:56]: I see. Thank you.

**Pull request merged**